shared/test/test_sources.rb

```diff
@@ -471,6 +471,121 @@ def test_restore_main_json_with_null_animation_versions
     assert_restores_main_json_with_animation_version nil
   end
 
+  def test_remix_source_file
+    # Mock destination
+    @destination_channel = create_channel
+    @destination_api = FilesApiTestHelper.new(current_session, 'sources', @destination_channel)
+    @destination_animation_api = FilesApiTestHelper.new(current_session, 'animations', @destination_channel)
+
+    animation_key = @api.add_random_suffix('animation-key')
+    animation_filename = "#{animation_key}.png"
+    delete_all_animation_versions(animation_filename)
+
+    # Create an animation
+    animation_v1 = 'stub-png-v1'
+    animation_v1_vid = put_animation(animation_filename, animation_v1)
+
+    # Update main.json
+    main_json_v1 = {
+      "source": "//version 1",
+      "animations": {
+        "orderedKeys": [animation_key],
+        "propsByKey": {
+          "#{animation_key}": {
+            "name": "Remix",
+            "sourceUrl": nil,
+            "version": animation_v1_vid
+          }
+        }
+      }
+    }.stringify_keys
+    put_main_json(main_json_v1)
+
+    # Remix
+    AnimationBucket.new.copy_files @channel, @destination_channel
+    SourceBucket.new.remix_source @channel, @destination_channel
```

Comment:
I'd really like to see _a_ test (maybe not this one?) that tests the `ProjectsController` API call at `/gamelab/<channel_id>/remix`. We'll probably need to do that through `ProjectsControllerTest` though.  Here, this is fine for now.

Reply:
I'm writing a remix test into the projects controller but I'll open that change in another PR.